pkg/ddevapp/envfile.go

```diff
@@ -27,7 +27,8 @@ func WriteProjectEnvFile(envFilePath string, envMap map[string]string, envText s
 	// envFilePath := filepath.Join(app.AppRoot, ".env")
 	for k, v := range envMap {
 		// Escape double quotes in the value, since we wrap it in double quotes
-		v = `"` + strings.ReplaceAll(v, `"`, `\"`) + `"`
+		// Escape $ in the value, since we don't want to expand it
+		v = `"` + strings.ReplaceAll(strings.ReplaceAll(v, `$`, `\$`), `"`, `\"`) + `"`
```

Comment:
I suspect this approach is intrusive and will have side-effects. It may be better to just link to the compose docs.

Reply:
This part of code is not related to `docker-compose`, this is what we do ourselves in `WriteProjectEnvFile`.

I already did several revisions of this code before, and I even tested `$` (but only in the end of the string, where it couldn't be treated as environment variable).

This PR handles `$` better, and makes `ddev dotenv set` work better with `$`.

Reply:
Thanks!

Reply:
Same comment on nested strings.ReplaceAll() here. Better as 2 steps.